src/user/newsletter/newsletter-subscription.controller.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import UserService from '../../services/user.service';
+import '@gravitee/ui-components/wc/gv-newsletter-subscription';
+
+class NewsletterSubscriptionController {
+
+  newsletterPage;
+
+  constructor(private $state,
+              private $scope,
+              public UserService: UserService,
+              private NotificationService,
+              Constants,
+              private $window,
+              private $rootScope) {
+    'ngInject';
+
+    $scope.user = UserService.currentUser;
+
+    this.newsletterPage = document.querySelector('gv-newsletter-subscription');
+    this.newsletterPage.addEventListener('gv-newsletter-subscription:subscribe', this.onSubscribe.bind(this));
+    this.newsletterPage.addEventListener('gv-newsletter-subscription:skip', this.onSkip.bind(this));
```

Comment:
If you want, the good syntax to add listeners in template is : 
`ng-on-gv-newsletter:subscribe="$ctrl.onSubscribe($event)"`

Reply:
as discussed, we will keep as is since this component is simple and we set a property directly here